dali/test/python/operator_1/test_clahe.py

```diff
@@ -0,0 +1,242 @@
+        num_threads=1,
+        device_id=0,
+        tiles_x=8,
+        tiles_y=8,
+        clip_limit=2.0,
+        bins=256,
+        luma_only=True,
+        input_shape=(128, 128, 1),
+    ):
+        super(ClahePipeline, self).__init__(batch_size, num_threads, device_id)
+        self.device = device
+        self.input_shape = input_shape
+        self.tiles_x = tiles_x
+        self.tiles_y = tiles_y
+        self.clip_limit = clip_limit
+        self.bins = bins
+        self.luma_only = luma_only
+
+    def define_graph(self):
+        # Create synthetic test data
+        data = fn.random.uniform(range=(0, 255), shape=self.input_shape, dtype=types.UINT8)
+
+        # Apply CLAHE
+        if self.device == "gpu":
+            data = data.gpu()
+
+        clahe_output = fn.clahe(
+            data,
+            tiles_x=self.tiles_x,
+            tiles_y=self.tiles_y,
+            clip_limit=self.clip_limit,
+            bins=self.bins,
+            luma_only=self.luma_only,
+        )
+
+        return data, clahe_output
+
+
+class ClaheOpsPipeline(Pipeline):
+    def __init__(
+        self,
+        device,
+        batch_size,
+        num_threads=1,
+        device_id=0,
+        tiles_x=8,
+        tiles_y=8,
+        clip_limit=2.0,
+        bins=256,
+        luma_only=True,
+        input_shape=(128, 128, 1),
+    ):
+        super(ClaheOpsPipeline, self).__init__(batch_size, num_threads, device_id)
+        self.device = device
+        self.input_shape = input_shape
+
+        self.clahe_op = ops.Clahe(
+            device=device,
+            tiles_x=tiles_x,
+            tiles_y=tiles_y,
+            clip_limit=clip_limit,
+            bins=bins,
+            luma_only=luma_only,
+        )
+
+    def define_graph(self):
+        # Create synthetic test data
+        data = fn.random.uniform(range=(0, 255), shape=self.input_shape, dtype=types.UINT8)
+
+        if self.device == "gpu":
+            data = data.gpu()
+
+        clahe_output = self.clahe_op(data)
+
+        return data, clahe_output
+
+
+def test_clahe_operator_registration():
+    """Test that CLAHE operator is properly registered."""
+    # Check functional API
+    assert hasattr(fn, "clahe"), "CLAHE operator not found in dali.fn"
+
+    # Check class API
+    assert hasattr(ops, "Clahe"), "CLAHE operator not found in dali.ops"
+
+    # Check schema
+    schema = dali.backend.TryGetSchema("Clahe")
+    assert schema is not None, "CLAHE schema not found"
+    assert schema.name == "Clahe"
+
+
+def test_clahe_grayscale_gpu():
+    """Test CLAHE with grayscale images on GPU."""
+    batch_size = 4
+    pipe = ClahePipeline(
+        "gpu", batch_size, input_shape=(64, 64, 1), tiles_x=4, tiles_y=4, clip_limit=2.0
+    )
+    pipe.build()
+
+    outputs = pipe.run()
+    input_data, clahe_output = outputs
+
+    # Verify output properties
+    assert len(clahe_output) == batch_size
+    for i in range(batch_size):
+        original = np.array(input_data[i])
```

Comment:
I think you need to add `.as_cpu()` before constructing the np.array. Otherwise it won't work in case of GPU outputs
```suggestion
        original = np.array(input_data[i].as_cpu())
```
Same applies to all conversions to np.array in this file.